static/app/components/profiling/flamegraphZoomView.tsx

```diff
@@ -782,4 +808,51 @@ const Canvas = styled('canvas')`
   position: absolute;
 `;
 
+// loosely based spreadsheet navigation
+const keyDirectionMap: Record<string, Direction> = {
+  ArrowUp: 'up',
+  ArrowDown: 'down',
+  ArrowLeft: 'left',
+  ArrowRight: 'right',
+  Tab: 'right',
+  'Shift+Tab': 'left',
+  Enter: 'down',
+  'Shift+Enter': 'up',
+  w: 'up',
+  s: 'down',
+  a: 'left',
+  d: 'right',
+};
+
+function handleFlamegraphKeyboardNavigation(
+  evt: KeyboardEvent,
+  currentFrame: FlamegraphFrame | undefined,
+  inverted: boolean = false
+) {
+  if (!currentFrame) {
+    return null;
+  }
+
+  const keyboardKey = !inverted
+    ? evt.key
+    : evt.key === 'ArrowUp'
+    ? 'ArrowDown'
+    : evt.key === 'ArrowDown'
+    ? 'ArrowUp'
+    : evt.key;
```

Comment:
You can drop negation and it may also be worth adding a comment here to explain as it might be hard for someone without a visual reference of the inversion to understand this

```suggestion
  const keyboardKey = inverted
    ? evt.key === 'ArrowUp' ? 'ArrowDown' : evt.key
    : evt.key === 'ArrowDown' ? 'ArrowUp' : evt.key
    : evt.key;
```

Reply:
your nested ternary code throws a syntax error.

in any event, i found this to be a naive we of handling it. i'm inverting specific keybinds, which is very dumb since any other key bound to up/down directions won't get the proper treatment.

i've remove this in favor of simply inverting `direction` when its either up/down

Reply:
>in any event, i found this to be a naive we of handling it. i'm inverting specific keybinds, which is very dumb since any other key bound to up/down directions won't get the proper treatment.

great move (and sorry about the suggestion, wish gh had some linting in the suggestion UI 😅 )